internal/jimm/access.go

```diff
@@ -532,6 +541,25 @@ func (t *tagResolver) groupTag(ctx context.Context, db *db.Database) (*ofga.Enti
 	return ofganames.ConvertTagWithRelation(entry.ResourceTag(), t.relation), nil
 }
 
+func (t *tagResolver) roleTag(ctx context.Context, db *db.Database) (*ofga.Entity, error) {
+	zapctx.Debug(
+		ctx,
+		"Resolving JIMM tags to Juju tags for tag kind: role",
+		zap.String("role-name", t.trailer),
+	)
+	if t.resourceUUID != "" {
+		return ofganames.ConvertTagWithRelation(jimmnames.NewRoleTag(t.resourceUUID), t.relation), nil
+	}
+	entry := dbmodel.RoleEntry{Name: t.trailer}
+
+	err := db.GetRole(ctx, &entry)
+	if err != nil {
+		return nil, errors.E(fmt.Sprintf("role %s not found", t.trailer))
```

Comment:
assumes a not found error..

Reply:
True... Let me address this in a follow up.